src/dmd/backend/aarray.d

```diff
@@ -0,0 +1,545 @@
+/**
+ * Compiler implementation of the
+ * $(LINK2 http://www.dlang.org, D programming language).
+ *
+ * Copyright:   Copyright (c) 2000-2018 by Digital Mars, All Rights Reserved
+ * Authors:     $(LINK2 http://www.digitalmars.com, Walter Bright), Dave Fladebo
+ * License:     Distributed under the Boost Software License, Version 1.0.
+ *              http://www.boost.org/LICENSE_1_0.txt
+ * Source:      https://github.com/dlang/dmd/blob/master/src/dmd/backend/aarray.d
+ */
+
+module dmd.backend.aarray;
+
+import core.stdc.stdio;
+import core.stdc.stdlib;
+import core.stdc.string;
+
+alias hash_t = size_t;
+
+struct aaA
```

Comment:
Looks like this can be private.

Reply:
* Does not follow standard D naming conventions
* Please come up with a better name and avoid abbreviations

Reply:
It's largely copy/pasted from the previous implementation in aa.c. I try to avoid renaming things in the first translated version, for reasons expounded upon several times.